src/pymmcore_plus/mda/metadata/schema.py

```diff
@@ -0,0 +1,206 @@
+    # and a 12-bit camera with a 16-bit buffer would be 2 bytes per pixel
+    bytes_per_pixel: int
+
+    # Number of components per pixel, RGBA is 4, GRAY is 1, etc...
+    # I think this will always be either 4 or 1
+    # note also that CMMCorePlus will fix BGRA to RGB, so the numpy array returned
+    # will have shape(..., 3) even though MMCore says components_per_pixel is 4
+    components_per_pixel: int  # rgb or not
+
+    # component_bit_depth is the "true" bit depth of the image, irrespective of the
+    # buffer size.  So a 12-bit gray camera will have 12 bits per component, and
+    # 1 components_per_pixel... even though the final numpy array will have
+    # dtype uint16 and bytes_per_pixel will be 2.
+    # An RGBA32 camera would have 8 bits per component and 4 components_per_pixel
+    component_bit_depth: int
+
+    # format of the pixel data, like "RGB32", "GRAY16", etc...
+    pixel_format: Literal["GRAY8", "GRAY16", "GRAY32", "RGB32", "RGB64", ""]
+
+    # name of the currently active pixel size configuration
+    pixel_size_config_name: str
+
+    # the product of magnification of all loaded devices of type MagnifierDevice
+    # If no devices are found, or all have magnification=1, this will not be present
+    magnification_factor: NotRequired[float]
+
+    # this will be != 1 for things like multi-camera device,
+    # or any "single" device adapter that manages multiple detectors, like PMTs, etc...
+    num_camera_adapter_channels: NotRequired[int]
+
+    pixel_size_um: float
+    # will not be present if equal to the default of (1,0,0,0,1,0)
+    pixel_size_affine: NotRequired[AffineTuple]
+
+    # will not be present if the ROI is the full image
+    roi: NotRequired[Tuple[int, int, int, int]]
+    # will not be present if camera devices not support multiple ROIs, or if multiple
+    # ROIs are not currently being used
+    multi_roi: NotRequired[Tuple[List[int], List[int], List[int], List[int]]]
+
+
+class Position(TypedDict):
+    """Represents a position in 3D space and focus."""
+
+    x: Optional[float]
+    y: Optional[float]
+    z: Optional[float]
+
+
+class PropertyValue(TypedDict):
+    """A single device property setting in a configuration group."""
+
+    dev: str
+    prop: str
+    val: Any
+
+
+class ConfigPreset(TypedDict):
+    """A group of device property settings."""
+
+    name: str
+    settings: Tuple[PropertyValue, ...]
+
+
+class PixelSizeConfigPreset(ConfigPreset):
+    """A specialized group of device property settings for a pixel size preset."""
+
+    pixel_size_um: float
+    pixel_size_affine: NotRequired[AffineTuple]
+
+
+class ConfigGroup(TypedDict):
+    """A group of configuration presets."""
+
+    name: str
+    presets: Tuple[ConfigPreset, ...]
+
+
+class SummaryMetaV1(TypedDict, total=False):
+    """Complete summary metadata for the system. Version 1.0."""
+
+    devices: Tuple[DeviceInfo, ...]
+    system_info: SystemInfo
+    image_infos: Tuple[ImageInfo, ...]
```

Comment:
Can you remind me how the correct index for this is determined for a given plane/frame? Or is that still TBD?

Reply:
TBD ... or, you would use the `camera_device` label in FrameMetadata to look up stuff in SummaryMetadata `(next(x for x in summary['image_infos'] if x.get('camera_label') == 'thing'))`